prelude.dx

```diff
@@ -323,45 +323,74 @@ def tabEq (n:Type) ?-> (eqA: Eq a) ?=> : Eq (n=>a) = MkEq $
 
 'Transcencendental functions
 
-data Floating a:Type =
-  MkFloating (a->a) (a->a) (a->a) (a->a) (a->a) (a->a) (a->a) (a->a) (a->a) (a->a) (a->a) (a->a) (a->a) (a->a->a) (a->a)
-
-def exp    (d:Floating a) ?=> : a -> a = case d of MkFloating f _ _ _ _ _ _ _ _ _ _ _ _ _ _ -> f
-def exp2   (d:Floating a) ?=> : a -> a = case d of MkFloating _ f _ _ _ _ _ _ _ _ _ _ _ _ _ -> f
-def log    (d:Floating a) ?=> : a -> a = case d of MkFloating _ _ f _ _ _ _ _ _ _ _ _ _ _ _ -> f
-def log2   (d:Floating a) ?=> : a -> a = case d of MkFloating _ _ _ f _ _ _ _ _ _ _ _ _ _ _ -> f
-def log10  (d:Floating a) ?=> : a -> a = case d of MkFloating _ _ _ _ f _ _ _ _ _ _ _ _ _ _ -> f
-def log1p  (d:Floating a) ?=> : a -> a = case d of MkFloating _ _ _ _ _ f _ _ _ _ _ _ _ _ _ -> f
-
-def sin    (d:Floating a) ?=> : a -> a = case d of MkFloating _ _ _ _ _ _ f _ _ _ _ _ _ _ _ -> f
-def cos    (d:Floating a) ?=> : a -> a = case d of MkFloating _ _ _ _ _ _ _ f _ _ _ _ _ _ _ -> f
-def tan    (d:Floating a) ?=> : a -> a = case d of MkFloating _ _ _ _ _ _ _ _ f _ _ _ _ _ _ -> f
-
-def floor  (d:Floating a) ?=> : a -> a = case d of MkFloating _ _ _ _ _ _ _ _ _ f _ _ _ _ _ -> f
-def ceil   (d:Floating a) ?=> : a -> a = case d of MkFloating _ _ _ _ _ _ _ _ _ _ f _ _ _ _ -> f
-def round  (d:Floating a) ?=> : a -> a = case d of MkFloating _ _ _ _ _ _ _ _ _ _ _ f _ _ _ -> f
-
-def sqrt   (d:Floating a) ?=> : a -> a = case d of MkFloating _ _ _ _ _ _ _ _ _ _ _ _ f _ _ -> f
-def pow    (d:Floating a) ?=> : a -> a -> a = case d of MkFloating _ _ _ _ _ _ _ _ _ _ _ _ _ f _ -> f
-
-def lgamma (d:Floating a) ?=> : a -> a = case d of MkFloating _ _ _ _ _ _ _ _ _ _ _ _ _ _ f -> f
+interface Floating a:Type where
+  floating_exp    : a -> a
+  floating_exp2   : a -> a
+  floating_log    : a -> a
+  floating_log2   : a -> a
+  floating_log10  : a -> a
+  floating_log1p  : a -> a
+  floating_sin    : a -> a
+  floating_cos    : a -> a
+  floating_tan    : a -> a
+  floating_floor  : a -> a
+  floating_ceil   : a -> a
+  floating_round  : a -> a
+  floating_sqrt   : a -> a
+  floating_pow    : a -> a -> a
+  floating_lgamma : a -> a
+
+def exp    (d:Floating a) ?=> : a -> a = floating_exp
+def exp2   (d:Floating a) ?=> : a -> a = floating_exp2
+def log    (d:Floating a) ?=> : a -> a = floating_log
+def log2   (d:Floating a) ?=> : a -> a = floating_log2
+def log10  (d:Floating a) ?=> : a -> a = floating_log10
+def log1p  (d:Floating a) ?=> : a -> a = floating_log1p
+def sin    (d:Floating a) ?=> : a -> a = floating_sin
+def cos    (d:Floating a) ?=> : a -> a = floating_cos
+def tan    (d:Floating a) ?=> : a -> a = floating_tan
+def floor  (d:Floating a) ?=> : a -> a = floating_floor
+def ceil   (d:Floating a) ?=> : a -> a = floating_ceil
+def round  (d:Floating a) ?=> : a -> a = floating_round
+def sqrt   (d:Floating a) ?=> : a -> a = floating_sqrt
+def pow    (d:Floating a) ?=> : a -> a -> a = floating_pow
+def lgamma (d:Floating a) ?=> : a -> a = floating_lgamma
```

Comment:
> Sorry of course I've misspelled in my previous review and I meant for `tanh` etc. to end up in `Floating`, not in `Fractional`. I'm happy to merge for now even without that change, but can you please rebase on top of the new tip of `dev` so that the CI fix I just push today lets us test this change?

Yes, I'll rebase, and I'll also try the shorter aliases you suggested again - they didn't work the first time I tried, but there were a few simultaneous problems.

Reply:
Surprisingly to me, the way to resolve this problem was to remove these function definitions entirely.  I hadn't realized that the members of every interface are all globally available already.

Reply:
Yeah! The whole point of the `interface` syntax is to generate all this boilerplate so you don't have to!